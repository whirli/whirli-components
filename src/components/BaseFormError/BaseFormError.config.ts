// @ts-ignore
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormError/BaseFormError.config';

// Package config
export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
