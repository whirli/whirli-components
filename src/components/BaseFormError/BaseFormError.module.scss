@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles
.error {}
// breakpoint styles
$packageBreakpointStyles: ();