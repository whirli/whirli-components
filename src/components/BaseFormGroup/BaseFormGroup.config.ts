// @ts-ignore
import type { PropType } from '@composition';

import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormGroup/BaseFormGroup.config';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
