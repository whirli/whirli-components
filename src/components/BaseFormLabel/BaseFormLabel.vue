<template>
  <label :for="props.name" :class="[classes, requiredClass]"><slot /></label>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormLabel/BaseFormLabel.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormLabel.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles.label, ...makeClasses(ComponentStyles, props, styles)];
const requiredClass: string = props.required ? styles.required : '';
</script>
