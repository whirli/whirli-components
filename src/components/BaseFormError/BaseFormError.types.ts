import { PropKeys, PropValues } from './BaseFormError.constants';
