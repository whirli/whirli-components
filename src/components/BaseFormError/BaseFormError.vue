<template>
  <BaseText v-bind="$attrs" :class="classes" v-if="$slots.default"><slot /></BaseText>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormError/BaseFormError.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormError.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
import BaseText from '@whirli-components/components/BaseText/BaseText.vue';
const { makeClasses } = useClasses();
const classes = [styles.error, ...makeClasses(ComponentStyles, ConfigProps, props, styles)];
</script>
