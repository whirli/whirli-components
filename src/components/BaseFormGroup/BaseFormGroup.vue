<template>
  <div :class="classes"><slot /></div>
</template>

<script setup lang="ts">
// Vue
;

// Styles
import styles from '@whirli-local/components/BaseFormGroup/BaseFormGroup.module.scss';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormGroup.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles['form-group'], ...makeClasses(ComponentStyles, props, styles)];
</script>
