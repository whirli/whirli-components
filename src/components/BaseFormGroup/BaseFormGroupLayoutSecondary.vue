<template>
  <BaseFormGroup>
    <BaseGrid gap="0">
      <BaseRow :direction="rowDirection">
        <BaseFormLabel :name="props.name">{{ props.label }}</BaseFormLabel> <slot />
      </BaseRow>
      <BaseRow v-if="props.error">
        <BaseFormError>{{ props.error }}</BaseFormError>
      </BaseRow>
    </BaseGrid>
  </BaseFormGroup>
</template>

<script setup lang="ts">
// Data
import { ConfigProps as SharedConfigProps } from './BaseFormGroupLayout.config';
import { ConfigProps } from './BaseFormGroupLayoutSecondary.config';

// Types
import { Props } from '@whirli-components/@types/props';
import { PropDirection } from './BaseFormGroup.types';

// @ts-ignore
const props: Props = defineProps({ ...SharedConfigProps, ...ConfigProps });

// Constants
import { PropKeys } from './BaseFormGroup.constants';

// Components
import BaseFormGroup from '@whirli-components/components/BaseFormGroup/BaseFormGroup.vue';
import BaseGrid from '@whirli-components/components/BaseGrid/BaseGrid.vue';
import BaseRow from '@whirli-components/components/BaseRow/BaseRow.vue';
import BaseFormLabel from '@whirli-components/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@whirli-components/components/BaseFormError/BaseFormError.vue';

const rowDirection: PropDirection = props[PropKeys.DIRECTION] === 'reverse' ? 'row-reverse' : 'row';
</script>
