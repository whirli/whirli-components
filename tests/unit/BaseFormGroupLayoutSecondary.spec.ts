import { mount } from '@vue/test-utils';
import BaseFormGroupLayoutSecondary from '@/components/BaseFormGroup/BaseFormGroupLayoutSecondary.vue';
import BaseFormGroup from '@/components/BaseFormGroup/BaseFormGroup.vue';
import BaseFormLabel from '@/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@/components/BaseFormError/BaseFormError.vue';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let wrapper: any;
const label = 'Hello World';
const content = 'Hello Content';
const error = 'Hello Error';

describe('BaseFormGroupLayoutSecondary.spec.vue', () => {
  beforeEach(() => {
    wrapper = mount(BaseFormGroupLayoutSecondary, {
      slots: {
        default: content,
      },
      props: {
        label,
        error,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has the FormGroup wrapper', () => {
    expect(wrapper.findComponent(BaseFormGroup).exists()).toBeTruthy();
  });

  it('label displays the correct text', () => {
    expect(wrapper.findComponent(BaseFormLabel).text()).toMatch(label);
  });

  it('error displays the correct text', () => {
    expect(wrapper.findComponent(BaseFormError).text()).toMatch(error);
  });
});
