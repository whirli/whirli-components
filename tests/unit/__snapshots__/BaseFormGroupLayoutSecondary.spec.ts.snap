// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormGroupLayoutSecondary.spec.vue renders the component 1`] = `
<div
  class=""
>
  
  <div
    class=""
  >
    
    <div
      class=""
    >
      
      
      <label
        class=""
      >
        
        Hello World
        
      </label>
       
      
      Hello Content
      
      
      
    </div>
    <div
      class=""
    >
      
      
      <span
        class=""
      >
        
        
        Hello Error
        
        
      </span>
      
      
    </div>
    
  </div>
  
</div>
`;
